mysql-test/suite/galera/t/galera_fk_delete.test

```diff
@@ -0,0 +1,40 @@
+#
+# This test verifies that, in a foreign key relationship, parallel applying of
+# parent table DELETE with child table DELETE is handled properly by the
+# applier threads.
+#
+--source include/have_innodb.inc
+--source include/galera_cluster.inc
+
+CREATE TABLE t1 (
+  id int primary key
+);
+
+CREATE TABLE t2 (
+  id int primary key ,
+  f_id int DEFAULT NULL, FOREIGN KEY(f_id)  REFERENCES t1 (id)
+);
+
+INSERT INTO t1 SELECT 1;
+
+# Ensure that node2 is configured with wsrep_slave_threads > 1
+--connection node_2
+--let $assert_text = node2 has parallel applier threads configured
+--let $assert_cond = [SELECT @@GLOBAL.wsrep_slave_threads > 1] = 1
+--source include/assert.inc
+
+--connection node_1
+# This makes DELETE on parent table to be executed in parallel with DELETE on child table.
```

Comment:
Would it be possible to use debug sync to be sure that deletion of the child table done by applier-1 is still in progress while applier-2 attempts to delete the parent table?
Right now it may happen that the child table is deleted before the applier even attempts to delete parent table.

Reply:
As discussed over slack, galera doesn't schedule parent table delete to execute parallely with child table delete with the fix
```
WITH FIX
--------
2021-01-05T11:21:29.248468Z 6 [Note] WSREP: cert debug: END CERTIFICATION (success): source: 1cd78202-4f48-11eb-9cd4-93f693744a89 version: 4 local: 0 state: CERTIFYING flags: 1 conn_id: 10 trx_id: 270 seqnos (l: 210, g: 205, s: 204, d: 204, ts: 20890921580973)
Trx seqno: 205
depends seqno:204

WITHOUT FIX
-----------
2021-01-05T11:42:20.380561Z 5 [Note] WSREP: cert debug: END CERTIFICATION (success): source: 06807f5e-4f4b-11eb-8f57-f3cf398b4153 version: 4 local: 0 state: CERTIFYING flags: 1 conn_id: 10 trx_id: 270 seqnos (l: 210, g: 205, s: 204, d: 203, ts: 22142053646410)
Trx seqno: 205
depends seqno:203
```
So, it ensures that the parent table delete gets executed only after the child table delete is completed.

Regarding the debug sync test case, I indeed came up with one such test [galera_fk_delete1.txt](https://github.com/percona/percona-xtradb-cluster/files/5770650/galera_fk_delete1.txt), but couldn't make it a 100% deterministic testcase due to timing issue in scheduling transactions. So, as discussed, I have updated the existing testcase to have more inserts on t2 and have also updated the comment.